src/routes/safe/components/SafeLoadError.tsx

```diff
@@ -1,20 +1,25 @@
-import { ReactElement, useEffect } from 'react'
-import { extractSafeAddress, WELCOME_ROUTE } from 'src/routes/routes'
-import { useDispatch } from 'react-redux'
+import { ReactElement } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+import { WELCOME_ROUTE } from 'src/routes/routes'
 import removeViewedSafe from 'src/logic/currentSession/store/actions/removeViewedSafe'
 import FetchError from './FetchError'
+import { currentSafeWithNames } from 'src/logic/safe/store/selectors'
 
 const SafeLoadError = (): ReactElement => {
   const dispatch = useDispatch()
 
-  useEffect(() => {
-    // Remove the errorneous Safe from the list of viewed safes on unmount
-    return () => {
-      dispatch(removeViewedSafe(extractSafeAddress()))
-    }
-  }, [dispatch])
+  const { address } = useSelector(currentSafeWithNames)
```

Comment:
If you only need the address you can also use the `currentSafe` selector directly.